lib/utils/compose_ts.ts

```diff
@@ -799,18 +771,18 @@ export async function tarDirectory(
 }
 
 /**
- * Add or override a `build-variables` section to the `.balena/balena.yml`
- * build metadata, with the contents of any `--buildArgs` flags provided
- * on the command line.
+ * Add or override a `build-variables` section in the `.balena/balena.yml`
+ * metadata file, with the contents of any `--buildArgs` flags provided
+ * on the command line (so that the command line flags take precedence).
  */
-function addBuildVariablesToBuildMetadata(
+function mergeMetadataFromFlags(
 	metadataPath: string,
-	metadata: MultiBuild.ParsedBalenaYml,
-	buildArgs: ParsedBuildArguments,
+	metadataFromFile: MultiBuild.ParsedBalenaYml,
+	metadataFromFlags: ParsedBuildArguments,
```

Comment:
Renamed to try to make it clearer: override "metadata from file" with "metadata from flags". This leaves the name `buildargs` to mostly refer to a simple dictionary that maps arg names to arg values, while "metadata" refers to the more complex object that includes the `services` and `global` properties.